csds-client/client/v3/client.go

```diff
@@ -198,20 +202,21 @@ func (c *ClientV3) doRequest(streamClientStatus csdspb_v3.ClientStatusDiscoveryS
 }
 
 // parseConfigStatus parses each xds config status to string
-func parseConfigStatus(xdsConfig []*csdspb_v3.PerXdsConfig) []string {
+func parseConfigStatus(xdsConfig []*csdspb_v3.ClientConfig_GenericXdsConfig) []string {
```

Comment:
Should we use PerXdsConfig **or** GenericXdsConfig to make it backward compatible? With current implementation, if a user is using an older version of Envoy, I would assume it would break, right?

@fuqianggao

Reply:
Config dump in xds_config (PerXdsConfig) is deprecated in csds v3: https://github.com/envoyproxy/envoy/blob/main/api/envoy/service/status/v3/csds.proto#L182. But, if we do plan to retain it for backward compatibility, then we need to make sure to populate xds_config in Traffic Director CSDS v3 API implementation.